build_pex.bash

```diff
@@ -0,0 +1,47 @@
+CACHE_ROOT="${XDG_CACHE_HOME:-$HOME/.cache}/pantsbuild-binaries"
+
+VIRTUALENV_VERSION="16.0.0"
+
+function ensure_venv_installed() {
+  local -r version="${VIRTUALENV_VERSION}"
+
+  local -r venv_cache_dir="${CACHE_ROOT}/virtualenv"
+  local -r venv_dir="${venv_cache_dir}/virtualenv-${version}"
+  if [[ ! -d "${venv_dir}" ]]; then
+    (
+      mkdir -p "${venv_cache_dir}"
+      cd "${venv_cache_dir}"
+      curl -LO https://pypi.python.org/packages/source/v/virtualenv/virtualenv-${version}.tar.gz >&2
+      tar -xvzf "virtualenv-${version}.tar.gz" >&2
+    )
+  fi
+  echo "${venv_dir}"
+}
+
+function activate_venv() {
+  local -r dir="$1"
+
+  local -r activate="${dir}/bin/activate"
+  if [[ ! -f "${activate}" ]]; then
+    mkdir -p "$(dirname "${dir}")"
+    (
+      cd "$(ensure_venv_installed)"
+      python2.7 virtualenv.py "${dir}"
+    )
+  fi 
+  source "${activate}"
+}
+
+PEX_VERSION="1.4.4"
```

Comment:
`pex` has long published itself as a pex - on the github releases page at e.g. `https://github.com/pantsbuild/pex/releases/download/v1.4.4/pex27`.

any reason to not just curl this single file bin at `$PEX_VERSION`, chmod and run vs setting up a venv just to get `pex(1)`?

Reply:
Nope! Being real dumb since I knew that / have done a handful of those releases. Will fix.